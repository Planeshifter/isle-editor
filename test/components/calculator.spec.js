// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from 'components/calculator';


// TESTS //

it( 'renders a basic calculator', () => {
	const calculator = renderer.create(
		<Calculator expandable={false} />
	).toJSON();
	expect( calculator ).toMatchSnapshot();
});

it( 'renders an expandable calculator', () => {
	const calculator = renderer.create(
		<Calculator expandable={true} />
	).toJSON();
	expect( calculator ).toMatchSnapshot();
});

it( 'renders an inactive calculator', () => {
	const calculator = renderer.create(
		<Calculator show={false} />
	).toJSON();
	expect( calculator ).toMatchSnapshot();
});
