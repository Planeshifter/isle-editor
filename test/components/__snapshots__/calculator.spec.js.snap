// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a basic calculator 1`] = `null`;

exports[`renders an expandable calculator 1`] = `
<div
  node={null}
>
  <div
    className="react-draggable"
    disabled={false}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    resizable={false}
    style={
      Object {
        "boxSizing": "border-box",
        "cursor": "move",
        "display": "inline-block",
        "flexShrink": 0,
        "height": "auto",
        "left": 0,
        "maxHeight": 9007199254740991,
        "maxWidth": 9007199254740991,
        "minHeight": undefined,
        "minWidth": undefined,
        "position": "fixed",
        "top": 0,
        "transform": "translate(30px,30px)",
        "userSelect": "auto",
        "width": "auto",
      }
    }
    transform={null}
  >
    <div
      className="panel  card"
      footer={null}
      header="calculator"
      id="calc-panel"
      role="button"
      style={
        Object {
          "height": undefined,
        }
      }
      tabIndex={0}
    >
      <h3
        className="card-header"
      >
        <span
          className="unselectable"
        >
          calculator
        </span>
      </h3>
      <div
        className="card-body"
        style={
          Object {
            "display": null,
          }
        }
      >
        <div
          className="desaturated container"
        >
          <input
            autoFocus={true}
            className="form-control"
            id="calc-text-area"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value="0"
          />
          <p>
            answer
             = 
            0
          </p>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <div
                className="row"
              >
                <button
                  className="input-button-small btn btn-light"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  (
                </button>
                <button
                  className="input-button-small btn btn-light"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  )
                </button>
                <button
                  className="input-button-small btn btn-warning"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  copy
                </button>
                <button
                  className="input-button-small btn btn-warning"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  expand
                </button>
              </div>
              <div
                className="row"
              >
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  7
                </button>
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  8
                </button>
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  9
                </button>
                <button
                  className="input-button-small btn btn-info"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  รท
                </button>
              </div>
              <div
                className="row"
              >
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  4
                </button>
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  5
                </button>
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  6
                </button>
                <button
                  className="input-button-small btn btn-info"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  ร
                </button>
              </div>
              <div
                className="row"
              >
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  1
                </button>
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  2
                </button>
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  3
                </button>
                <button
                  className="input-button-small btn btn-info"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  -
                </button>
              </div>
              <div
                className="row"
              >
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Ans
                </button>
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  0
                </button>
                <button
                  className="input-button-small btn btn-dark"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  .
                </button>
                <button
                  className="input-button-small btn btn-info"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  +
                </button>
              </div>
              <div
                className="row"
              >
                <button
                  className="input-button-small btn btn-danger"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  CE
                </button>
                <button
                  className="input-button-small btn btn-danger"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  AC
                </button>
                <button
                  className="half-button btn btn-success"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  =
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span />
  </div>
</div>
`;

exports[`renders an inactive calculator 1`] = `null`;
